Automattic/pocket-casts-ios
# Apple SSO: Watch token refresh

| 📘 Project: #381 |
|:---:|

| 📓 In my testing, Apple SSO didn't always return the email in the simulator, so testing with a physical device is recommended. |
|:---:|

Syncs the user login tokens to the watch to preserve session management. 

## To test

### Setup
1. Enable the Feature Flag `signInWithApple`
2. Set your scheme to `Staging` 

### Testing
You'll want to perform these tests with Username and Password, and Apple SSO as the flow changes touched each path.

1. Login on the App
2. Open the watch app 
3. Trigger an account refresh if the app doesn't trigger one itself
    - The app can be slow to send messages on builds from Xcode so be patient. 
4. Expect to be logged in to the same account as the phone.
5. Log out and try the other path.  

## Checklist
